File: WizQuixTwitchPaintExtension/Classes/Models/Commands/SetPixelCommand.ts
class SetPixelCommand extends ACommand
{
    public constructor(client:WebClient)
    {
        super("setpixel", client);
    }

    public OnInfo(data: string[])
    {
        if(data.length == 1 && data[0] == "OK") return;
        if(data.length < 3) return;

        let nickname = data[0];
        let coordStr = data[1];
        let colorname = data.slice(2).join(" ");

        let coords = this.Client.Canvas.ParseCoordinate(coordStr);
        if(coords === undefined) return;

        let color = this.Client.Canvas.Palette.GetColorByName(colorname);
        if(color === undefined) return;

        this.Client.Canvas.SetPixel(coords.x, coords.y, color, true);
    }

    public OnError(code: number, error: string)
    {
        
    }
}

File: WizQuixTwitchPaintApiService/Logic/Commands/SetPixelCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WizQuixTwitchPaintApiService.Logic.Commands
{
    /// command: setpixel <coordinate:string> <colorname:string> // coordinate in formal <letter:char><number:int>
    /// on success: info setpixel OK
    /// on fail:
    ///    - error setpixel 404 Color '<colorname:string>' didn't exists
    ///    - error setpixel 400 Coordinate is out of range
    ///    - error setpixel 400 Wrong number of arguments 
    /// examples:
    ///    - setpixel A1 Red
    ///    - setpixel d13 blue
    public class SetPixelCommand : ACommand
    {
        public SetPixelCommand(WebClient client) : base("SetPixel", client) { }

        public override async Task Execute(string[] args)
        {
            if(args.Length < 2)
            {
                await SendError(WebClient.HttpCodes.Code400, "Wrong number of arguments");
                return;
            }

            var colorname = string.Join(" ", args.Skip(1).ToArray());
            var res = await Client.JoinedRoom.SetPixel(Client, args[0], colorname, false);
            switch(res)
            {
                case Room.SetPixelErrors.ColorDidntExists:
                    await SendError(WebClient.HttpCodes.Code404, $"Color '{args[1]}' didn't exists");
                    break;
                case Room.SetPixelErrors.OutOfRange:
                    await SendError(WebClient.HttpCodes.Code400, "Coordinate is out of range");
                    break;
                case Room.SetPixelErrors.OK:
                    await SendInfo();
                    break;
            }
        }
    }
}
